core/server/proxy/src/main/java/alluxio/proxy/s3/signature/AwsSignatureProcessor.java

```diff
@@ -50,18 +53,41 @@ public AwsSignatureProcessor(ContainerRequestContext context) {
     mContext = context;
   }
 
+  /**
+   * Create a new {@link AwsSignatureProcessor} with HttpServletRequest
+   * as the info marshall source.
+   * Used by the new architecture in {@link alluxio.proxy.s3.S3RequestServlet}
+   *
+   * @param request
+   */
+  public AwsSignatureProcessor(HttpServletRequest request) {
+    mServletRequest = request;
+  }
+
   /**
    * Extract signature info from request.
    * @return SignatureInfo
    * @throws S3Exception
    */
   public SignatureInfo parseSignature() throws S3Exception {
-    Map<String, String> headers = S3RestUtils.fromMultiValueToSingleValueMap(
-        mContext.getHeaders(), true);
-    String authHeader = headers.get(AUTHORIZATION);
-    String dateHeader = headers.get(S3_SIGN_DATE);
-    Map<String, String> queryParameters = S3RestUtils.fromMultiValueToSingleValueMap(
-            mContext.getUriInfo().getQueryParameters(), false);
+    Map<String, String> queryParameters;
+    String authHeader;
+    String dateHeader;
+    if (mContext != null) {
+      Map<String, String> headers = S3RestUtils.fromMultiValueToSingleValueMap(
+              mContext.getHeaders(), true);
+      authHeader = headers.get(AUTHORIZATION);
+      dateHeader = headers.get(S3_SIGN_DATE);
+      queryParameters = S3RestUtils.fromMultiValueToSingleValueMap(
+              mContext.getUriInfo().getQueryParameters(), false);
+    } else {
```

Comment:
wondering whether a return to avoid else

Reply:
you mean return after line82? there's still processing after the if else section

Reply:
line82, then the else branch can be removed.

Reply:
return after line82 and remove line 83 to 90 ? there's still code section after 90